### DISCUSSION TITLE
"DFN + Current collector (potential pair)" model gives an error when a rest is included in the pybamm experiment

### DISCUSSION THREAD

**Discussion by abhishek-appana** [pybamm-team/PyBaMM#4555] [Q&A]
Hello everyone,

Below is a minimum working example for the simulation I am trying to run! It is a simple DFN + Current collector simulation using a custom experiment. I am able to run the simulation with either charge/discharge phases but I get an error (given below) when I include rest in the experiment. Has anyone faced this earlier? Is there a solution to resolve this?

```python
import pybamm as pb

pb.set_logging_level("INFO")
options = {"dimensionality": 2, "current collector": "potential pair"}
model = pb.lithium_ion.DFN(options=options)

parameter_values = pb.ParameterValues("Ecker2015")
parameter_values["Initial concentration in negative electrode [mol.m-3]"] = 0.0385*parameter_values["Maximum concentration in negative electrode [mol.m-3]"]
parameter_values["Initial concentration in positive electrode [mol.m-3]"] = 0.9517*parameter_values["Maximum concentration in positive electrode [mol.m-3]"]

experiment = pb.Experiment(
    [
        (
            "Rest for 60 seconds",
            "Charge at 1C for 1 hours or until 4.2 V",
            # "Discharge at 1C for 1 hours or until 3 V"
        )
    ]
)

solver = pb.CasadiSolver(mode="safe",dt_max=10)
sim = pb.Simulation(model, parameter_values=parameter_values, experiment=experiment, solver=solver)
sim.solve([0, 3600])
sim.plot()
``` 

Error message:
<details><summary>Details</summary>
<p>

---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
File Untitled-1:23
     [21](untitled-1:21) solver = pb.CasadiSolver(mode="safe",dt_max=10)
     [22](untitled-1:22) sim = pb.Simulation(model, parameter_values=parameter_values, experiment=experiment, solver=solver)
---> [23](untitled-1:23) sim.solve([0, 3600])
     [24](untitled-1:24) sim.plot()

NotImplementedError: Shape not recognized for 31920.0 * y[0:20000](note processing of 3D variables is not yet implemented)

</p>
</details>

**Comment by rtimms** [comment]
See https://github.com/pybamm-team/PyBaMM/issues/1549

**Reply by abhishek-appana** [reply]
Thank you for your response @rtimms.

I think there's an easier/simpler way around this. Instead of using a custom experiment like in the MWE above, I created a .csv file with the current and assigned that to the current function (as below).

```python
parameter_values["Current function [A]"] = current_interpolant
``` 

current_interpolant is the output of _pybamm.interpolant_ using _pybamm.t_
It worked!
